src/main/java/hudson/plugins/ec2/EC2Cloud.java

```diff
@@ -1186,32 +1214,30 @@ protected FormValidation doTestConnection(@AncestorInPath ItemGroup context, URL
                 return FormValidation.ok();
             }
             try {
-                SSHUserPrivateKey sshCredential = getSshCredential(sshKeysCredentialsId, context);
-                String privateKey = "";
-                if (sshCredential != null) {
-                    privateKey = sshCredential.getPrivateKey();
-                } else {
-                    return FormValidation.error("Failed to find credential \"" + sshKeysCredentialsId + "\" in store.");
-                }
-
                 AWSCredentialsProvider credentialsProvider = createCredentialsProvider(useInstanceProfileForCredentials, credentialsId, roleArn, roleSessionName, region);
                 AmazonEC2 ec2 = AmazonEC2Factory.getInstance().connect(credentialsProvider, ec2endpoint);
-                ec2.describeInstances();
```

Comment:
Why is this being deleted? IIUC this was an important part of form validation.

Reply:
is that because it proves `ec2` policy is correctly defined?